### PR TITLE
no support endpointslice in linux userpace mode 

### PR DESCRIPTION
**Pull request by JornShen**
/kind bug

in userpace mode of linux, we have not implement endpointslice resource processing logic.
we need to watch endpoint as before in this mode.

Ref https://github.com/kubernetes/kubernetes/issues/100880#

```release-note
no support endpointslice in linux userpace mode 
```

### CODE REVIEW

**Comment by JornShen** [comment]
/cc @robscott  @aojea 

I saw the enhancement
https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/0752-endpointslices/README.md

And In `Motivation`, there is words as below. 
```
For a K8s service, if its Endpoints object is too large, endpoint updates will not be propagated to kube-proxy(s), and thus iptables/ipvs won’t be reprogrammed.
```
EndpointSlice implement only in ipvs mode and iptables mode.
So I think we dont need to implemet userspace. In userspace mode, it will wirte too much rule to host.  
right ?

**Comment by aojea** [comment]
/sig network
/assign @danwinship 
please, let's hold on it a bit, we were discussing this recently and it didn't come to a conclusion yet

**Comment by danwinship** [comment]
If someone volunteered to do this work, we'd review it, but I don't think anyone cares about doing it proactively. The userspace proxy is mostly-deprecated.

**Comment by JornShen** [comment]
> If someone volunteered to do this work, we'd review it, but I don't think anyone cares about doing it proactively. The userspace proxy is mostly-deprecated.

@danwinship 
How about doing like this MR ? Just still watch the endpoints.  userspace mode doesn't  write much rules to iptables. And I dont think this mode must implement the endpointslice watching.

**Comment by aojea** [comment]
/hold cancel
/retest
it lgtm but I defer to Dan last decision

**Comment by JornShen** [comment]
ping @danwinship

File: cmd/kube-proxy/app/server.go
Comment (danwinship) [inline cmd/kube-proxy/app/server.go:739]: I think this logic belongs in `server_others.go` (and `server_windows.go`); it should not set `UseEndpointSlices` if `ProxyMode` is a mode that doesn't support it.

**Review by danwinship — CHANGES_REQUESTED** [review]
oh, sorry, I missed that this was a bugfix

**Comment by danwinship** [comment]
/remove-sig network
/sig network
(trying to fix the "do-not-merge/needs-sig" label)

File: cmd/kube-proxy/app/server.go
Comment (JornShen) [reply cmd/kube-proxy/app/server.go]: ohh, sorry, updated.

**Comment by JornShen** [comment]
/cc @danwinship 
PTAL :)

**Review by danwinship — CHANGES_REQUESTED** [review]
can you update `server_windows.go` too?

**Comment by JornShen** [comment]
> can you update `server_windows.go` too?

@danwinship 
updated. PTAL :)

**Comment by puerco** [comment]
@JornShen will you cherry pick this to 1.18?

**Comment by JornShen** [comment]
@puerco thanks for reminding me.
and please see https://github.com/kubernetes/kubernetes/pull/101741
@danwinship

**Pull request merged**
